frontend-v2/src/app/interest/generator/generator-form.tsx

```diff
@@ -0,0 +1,264 @@
+    params.append('name', data.name)
+    params.append('title', data.title)
+    params.append('description', data.description)
+    params.append('chapterId', data.chapterId)
+    params.append('showInterests', data.activismInterests.toString())
+    params.append(
+      'referralApply',
+      data.referralApplyName === '' ? 'null' : data.referralApplyName,
+    )
+    params.append('showReferralFriends', data.referralFriends.toString())
+    params.append('showReferralApply', data.referralApply.toString())
+    params.append('showReferralOutlet', data.referralOutlet.toString())
+
+    const url = `https://adb.dxe.io/interest?${params.toString()}`
+    setOutput(url)
+  }
+
+  return (
+    <div className="w-full max-w-2xl mx-auto p-8 bg-white rounded shadow">
+      <Form {...form}>
+        <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-6">
+          <FormField
+            control={form.control}
+            name="name"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Form name (visible in ADB)</FormLabel>
+                <FormControl>
+                  <Input type="text" placeholder="Name" {...field} />
+                </FormControl>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="title"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Form title</FormLabel>
+                <FormControl>
+                  <Input type="text" placeholder="Title" {...field} />
+                </FormControl>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="description"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Form description</FormLabel>
+                <FormControl>
+                  <Textarea placeholder="Description" {...field} />
+                </FormControl>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="chapterId"
+            rules={{ required: true }}
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>
+                  Chapter <span className="text-red-500">*</span>
+                </FormLabel>
+
+                <Select
+                  key={field.value}
+                  value={field.value}
+                  onValueChange={field.onChange}
+                  disabled={chapterListIsLoading}
+                >
+                  <FormControl>
+                    <SelectTrigger>
+                      <SelectValue placeholder="Select a chapter" />
+                    </SelectTrigger>
+                  </FormControl>
+                  <SelectContent>
+                    {chapterList &&
+                      chapterList.map((chapter: any) => (
```

Comment:
❗ no `any` pls when it can be avoided (ideally our eslint rules prevent this)
```suggestion
                      chapterList.map((chapter) => (
```